src/server/admin/models.py

```diff
@@ -35,97 +35,85 @@ def __init__(self, api_key: str, email: str = None) -> None:
         self.api_key = api_key
         self.email = email
 
-    @staticmethod
-    def list_users() -> List["User"]:
-        with Session() as session:
-            return session.query(User).all()
-
     @property
     def as_dict(self):
         return {
             "id": self.id,
             "api_key": self.api_key,
             "email": self.email,
-            "roles": User.get_user_roles(self.id),
+            "roles": set(role.name for role in self.roles),
             "created": self.created,
             "last_time_used": self.last_time_used
         }
 
-    @staticmethod
-    def get_user_roles(user_id: int) -> Set[str]:
-        with Session() as session:
-            user = session.query(User).filter(User.id == user_id).first()
-            return set([role.name for role in user.roles])
-
     def has_role(self, required_role: str) -> bool:
-        return required_role in User.get_user_roles(self.id)
+        return required_role in set(role.name for role in self.roles)
```

Comment:
maybe?  probably?

you caught me being a little lazy: the old `get_user_roles()` call that this replaced returned a set, and i just duplicated that behavior.  we do need a comprehension of some sort though, because we need to pull `.name` out of each role.  would this be better?  im not sure:

```suggestion
        return required_role in [role.name for role in self.roles]
```

Reply:
Oh I missed the old set return. Hmm. In that case I'd leave it as a `set`, and note for the future that if we ever need a user to be in dozens of roles we should convert to a for loop with an early exit.